extensions/wikia/Chat2/ChatEntryPoint.class.php

```diff
@@ -23,35 +23,56 @@ static public function onParserFirstCallInit( Parser &$parser ) {
 	}
 
 	/**
-	 * @param $isEntryPoint - set to true for chat entry point embeded in the article, false for rail module
-	 * Return an array of variables needed to render chat entry point template
+	 * Return an array of variables needed to render chat entry point mustache template
+	 *
+	 * @param $fromParserTag - set to true for chat entry point embedded in the article,
+	 * set to false for rail module
 	 *
 	 * @return array
 	 */
-	static public function getEntryPointTemplateVars( $isEntryPoint ) {
-		global $wgEnableWallExt, $wgBlankImgUrl;
-		return [
-			'linkToSpecialChat' => SpecialPage::getTitleFor( "Chat" )->escapeLocalUrl(),
-			'isEntryPoint' => $isEntryPoint,
+	static public function getEntryPointTemplateVars( $fromParserTag ) {
```

Comment:
I don't see a need to create a separate `$mode` var, but defining classnames here, instead of putting logic in the template is a great idea.